pravega/bookkeeper-operator
# BookKeeper Operator API Resource Hook.

### Change log description

* Adds a manifest which defines a post-install/upgrade hook to make sure the `bookkeepercluster` resource exists.

### Purpose of the change

* If one is unlucky with the timings when install the `bookkeeper-operator` chart, directly followed by the `bookkeeper` chart, the api-resource `BookKeeperCluster` may not exist by the time the bookkeeper chart tries to create a `BookKeeperCluster`.

### What the code does

* See above.

### How to verify it

Run many install/uninstall cycles of the mentioned charts -- see that the bookkeeper chart does not start before its resource dependency is registered.
